prost/src/encoding.rs

```diff
@@ -721,6 +721,128 @@ fixed_width!(
     get_i64_le
 );
 
+pub mod enumeration {
+    use super::*;
+    use crate::OpenEnum;
+
+    pub fn encode<T, B>(tag: u32, value: &OpenEnum<T>, buf: &mut B)
+    where
+        T: Clone + Into<i32>,
+        B: BufMut,
+    {
+        int32::encode(tag, &value.to_raw(), buf)
+    }
+
+    pub fn merge<T, B>(
+        wire_type: WireType,
+        value: &mut OpenEnum<T>,
+        buf: &mut B,
+        _ctx: DecodeContext,
+    ) -> Result<(), DecodeError>
+    where
+        i32: TryInto<T>,
+        B: Buf,
+    {
+        let mut raw = 0;
+        int32::merge(wire_type, &mut raw, buf, DecodeContext::default())?;
+        *value = OpenEnum::from_raw(raw);
+        Ok(())
+    }
+
+    pub fn encode_repeated<T, B>(tag: u32, values: &[OpenEnum<T>], buf: &mut B)
+    where
+        T: Clone + Into<i32>,
+        B: BufMut,
+    {
+        for value in values {
+            encode(tag, value, buf);
+        }
+    }
+
+    pub fn encode_packed<T, B>(tag: u32, values: &[OpenEnum<T>], buf: &mut B)
+    where
+        T: Clone + Into<i32>,
+        B: BufMut,
+    {
+        if values.is_empty() {
+            return;
+        }
+
+        encode_key(tag, WireType::LengthDelimited, buf);
+        let len: usize = values
+            .iter()
+            .map(|value| encoded_len_varint(value.to_raw() as u64))
+            .sum();
+        encode_varint(len as u64, buf);
+
+        for value in values {
+            encode_varint(value.to_raw() as u64, buf);
+        }
```

Comment:
Could `int32::encode_packed` be reused here?

Reply:
Same here, not worth changing the signature of `int32::encode_packed` for this.